noir-projects/aztec-nr/aztec/src/macros/mod.nr

```diff
@@ -216,6 +248,26 @@ comptime fn generate_process_log() -> Quoted {
     for i in 0..notes.len() {
         let (typ, (_, packed_note_length, _, _)) = notes[i];
 
+        // We get the typed expressions of `get_note_type_id` and `unpack` here rather than directly inlining them
+        // in the quote with `$typ::get_note_type_id()` and `$typ::unpack()`. Direct inlining would result
+        // in missing import warnings in the generated code (specifically, warnings that NoteInterface and
+        // Packable<N> are implemented but not in scope).
+        let get_note_type_id = typ.get_trait_impl(NOTE_INTERFACE).unwrap().methods().filter(|m| {
+            m.name() == quote { get_note_type_id }
+        })[0]
+            .as_typed_expr();
+
+        // We obtain the packable trait constraint here rather than storing it as a global variable. This is because
+        // the trait has a generic parameter that gets matched on first use. If we used a global variable, we would
+        // fail to get trait implementations for types that don't match the already-matched generic parameter.
+        let packable: TraitConstraint =
+            quote { crate::protocol_types::traits::Packable<_> }.as_trait_constraint();
+
+        let unpack = typ.get_trait_impl(packable).unwrap().methods().filter(|m| {
+            m.name() == quote { unpack }
+        })[0]
+            .as_typed_expr();
+
```

Comment:
Neat! I wonder if we could extract this into a function that we can reuse, though we'd need to pass `Quoted` values to it.

Reply:
Good idea. Added it in the "get_trait_impl_method util" commit.